src/fw-logs/fw-logs-parser.cpp

```diff
@@ -1,90 +1,215 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
-#include "fw-logs-parser.h"
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <src/fw-logs/fw-logs-parser.h>
+#include <src/fw-logs/fw-string-formatter.h>
+#include <src/fw-logs/fw-logs-xml-helper.h>
+
+#include <rsutils/string/from.h>
+
 #include <regex>
-#include <sstream>
-#include "fw-string-formatter.h"
-#include "stdint.h"
+#include <fstream>
+#include <iterator>
 
-using namespace std;
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        fw_logs_parser::fw_logs_parser(string xml_content)
-            : _fw_logs_formating_options(xml_content),
-            _last_timestamp(0),
-            _timestamp_factor(0.00001)
+        fw_logs_parser::fw_logs_parser( const std::string & definitions_xml )
+            : _source_to_formatting_options()
         {
-            _fw_logs_formating_options.initialize_from_xml();
+            fw_logs_xml_helper helper;
+            _source_id_to_name = helper.get_listed_sources( definitions_xml );
+            for( auto & source : _source_id_to_name )
+            {
+                std::string path = helper.get_source_parser_file_path( source.first, definitions_xml );
+                std::ifstream f( path.c_str() );
+                if( f.good() )
+                {
+                    std::string xml_contents;
+                    xml_contents.append( std::istreambuf_iterator< char >( f ), std::istreambuf_iterator< char >() );
+                    fw_logs_formatting_options format_options( std::move( xml_contents ) );
+                    _source_to_formatting_options[source.first] = format_options;
```

Comment:
Can you explain the name `_source_to_formatting_options`?

It holds the relevant files path right?

Reply:
This maps source (by id) to `fw_logs_formatting_options`.
The `fw_logs_formatting_options` class holds the data needed to build the log message from the binary data - event format strings, file names etc...

Reply:
I used the old name, but perhaps renaming this class with `data_base` rather than `options` is better understood, as it just holds data parsed from the file.